File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/mediator/Tester.java
package mediator;

class Tester extends Colleague {
    public Tester(Team team) {
        super(team);
    }

    @Override
    public void send(String message) {
        System.out.println("Tester sending message: " + message);
        team.request(message, this);
    }

    @Override
    public void receive(String message) {
        System.out.println("Tester received task: " + message);
    }
}

File: code/csharp/DesignPatterns/mediator/Tester.cs
namespace Mediator;

class Tester : Colleague
{
    public Tester(IMediator mediator) : base(mediator) { }

    public override void Send(string message)
    {
        System.Console.WriteLine($"Tester sending message: {message}");
        Mediator.Request(message, this);
    }

    public override void Receive(string message)
    {
        System.Console.WriteLine($"Tester received task: {message}");
    }
}
